src/main/java/com/aws/greengrass/builtin/services/lifecycle/LifecycleIPCEventStreamAgent.java

```diff
@@ -304,75 +319,100 @@ public List<Future<DeferComponentUpdateRequest>> sendPreComponentUpdateEvent(
+                componentUpdateListenersInternal.entrySet()) {
             String serviceName = entry.getKey();
-            entry.getValue().forEach(subscribeHandler -> {
-                ComponentUpdatePolicyEvents events = makePreUpdateEvents(serviceName, preComponentUpdateEvent);
-
-                CompletableFuture<DeferComponentUpdateRequest> deferUpdateFuture = new CompletableFuture<>();
-                // If there are multiple pre component events sent to same service, we will store the latest future
-                // As the update should be waiting for the latest one to complete.
-                Pair<String, String> serviceAndDeployment =
-                        new Pair<>(serviceName, preComponentUpdateEvent.getDeploymentId());
-                // Save to the map before sending the event so that it will be there if
-                // they respond very quickly
-                deferUpdateFuturesMap.put(serviceAndDeployment, deferUpdateFuture);
-
-                try {
-                    subscribeHandler.accept(events);
-                } catch (Exception e) {
-                    log.atError().setCause(e).kv(COMPONENT_NAME, serviceName)
-                            .log("Failed to send the pre component update on stream");
-                    deferUpdateFuturesMap.remove(serviceAndDeployment);
-                    return;
-                }
-                deferUpdateFutures.add(deferUpdateFuture);
-            });
+            entry.getValue()
+                    .forEach(
+                            subscribeHandler -> {
+                                ComponentUpdatePolicyEvents events =
+                                        makePreUpdateEvents(serviceName, preComponentUpdateEvent);
+
+                                CompletableFuture<DeferComponentUpdateRequest> deferUpdateFuture =
+                                        new CompletableFuture<>();
+                                // If there are multiple pre component events sent to same service,
+                                // we will store the latest future
+                                // As the update should be waiting for the latest one to complete.
+                                Pair<String, String> serviceAndDeployment =
+                                        new Pair<>(
+                                                serviceName,
+                                                preComponentUpdateEvent.getDeploymentId());
+                                // Save to the map before sending the event so that it will be there
+                                // if
+                                // they respond very quickly
+                                deferUpdateFuturesMap.put(serviceAndDeployment, deferUpdateFuture);
+
+                                try {
+                                    subscribeHandler.accept(events);
+                                } catch (Exception e) {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

It appears that your code handles a broad swath of exceptions in the catch block, potentially trapping dissimilar issues or problems that should not be dealt with at this point in the program.

[Learn more](https://cwe.mitre.org/data/definitions/396.html)